src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -120,12 +120,24 @@ case class CsvRelation protected[spark] (
       schemaFields: Seq[StructField],
       projection: MutableProjection,
       row: GenericMutableRow): Iterator[Row] = {
-    iter.map { line =>
-      val tokens = CSVParser.parse(line, csvFormat).getRecords.head
-      for (index <- 0 until schemaFields.length) {
-        row(index) = tokens.get(index)
+    iter.flatMap { line =>
+      try {
+        val records = CSVParser.parse(line, csvFormat).getRecords
+        if (records.isEmpty) {
+          logger.warn(s"Ignoring empty line: $line")
+          None
+        } else {
+          val tokens = records.head
```

Comment:
Right now we expect that each element of the RDD is a line, but if we just iterated over the records and returned them, would that make everyone happy? Could also avoid the initial if check.